### PR TITLE
Mirate/feat/2 implement tcp socket lifecycle and poll() event handlers

### LINKED ISSUE #2: Implement TCP socket lifecycle and poll() events handler
set up the socket lifecycle and poll() events handler

### PR DESCRIPTION
**Pull request by miouz**
## feat: TCP socket lifecycle and poll() event handlers

Sets up the base server socket so we can accept connections and call the right event handlers ( Client or Server)

Whats in:

1. it creats Servers ( for now i set up 5 servers to test the multi-servers) and setUpServers with socket(), listen() and bind()
2. it sets fds to non blocking
3. it sets up a loop with poll() in order to watch on all events arriving on fds of the program
4. when poll() returns positif number N, it means N fds got events:
5. if is POLLIN , it calls the events handler according to if is a client or server's fd
6. if is POLLOUT, it send the response to the client
7. in case of poll() error, it exists
8. other server or client erros it will just clean resources and continue on next event

**To test:** `make debug`  then run and hit `localhost:8080` , `localhost:6060`, `localhost:5050`, `localhost:4040`, `localhost:3030`in a browser, you should see a message of recieved request in the terminal when a connection is accepted.

several edge cases to test: 

1.Test timeout — connect and do nothing for 30s 
2.Test multiple clients simultaneously
for i in {1..5}; do nc localhost 8080 & done
3.valgrind --leak-check=full --track-origins=yes ./webserv
Then connect/disconnect clients with Ctrl+C after nc the server

Closes: #2

### CODE REVIEW

File: Server.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+		throw std::runtime_error(std::string("fcntl: ") + strerror(errno));
+
+	address_.sin_family = AF_INET;
+	address_.sin_addr.s_addr = INADDR_ANY;
+	address_.sin_port = htons(port_);
+	if (bind(sockFd_, (struct sockaddr*)&address_, sizeof(address_)) == RETURN_ERROR)
+		throw std::runtime_error(std::string("bind: ") + strerror(errno));
+
+
+	if (listen(sockFd_, 128) == RETURN_ERROR)
```
Comment (sheikice) [inline Server.cpp]: What is this 128 magic number ?

Comment (miouz) [reply Server.cpp]: That number is the backlog — it tells the OS how many incoming connections to queue up while your server is busy and hasn't called accept() yet.
It's not the maximum number of simultaneous clients, just the size of the waiting room before the OS starts refusing new connections.
Why 128 specifically? It's a common convention. On Linux, the actual value is silently capped by the kernel at whatever net.core.somaxconn is set to, which is typically 128 by default anyway. So passing anything higher doesn't do much unless you've tuned that kernel parameter.
Some codebases use SOMAXCONN as the constant

Comment (sheikice) [reply Server.cpp]: Ok, is it possible  to replace this number with a constant name ?

Comment (miouz) [reply Server.cpp]: job done

**Review by ceslinge — APPROVED** [review]
Our time is ruuuuuuuuuuniing ouuuut

**Review by sheikice — APPROVED** [review]
Incroyable, le code est parfait ! Tu as mon autorisation pour merge ce PR 👍
